compositor_render/src/scene/view_component/layout.rs

```diff
@@ -175,6 +188,32 @@ impl ViewComponentParam {
         )
     }
 
+    fn scale_factor_for_overflow_fit(
+        &self,
+        size: Size,
+        children: &[StatefulComponent],
+        pts: Duration,
+    ) -> f32 {
+        let sum_size = self.sum_static_children_sizes(children, pts);
+        let (max_size, max_alternative_size) = match self.direction {
+            super::ViewChildrenDirection::Row => (size.width, size.height),
+            super::ViewChildrenDirection::Column => (size.height, size.width),
+        };
+        let max_alternative_size_for_child = Self::static_children_iter(children, pts)
```

Comment:
If direction is row then I consider size to be widht and alternative size to be height. I couldn't find a better name, I'm open to suggestions